src/App/Part.cpp

```diff
@@ -109,6 +109,23 @@ PyObject *Part::getPyObject()
     return Py::new_reference_to(PythonObject);
 }
 
+void Part::handleChangedPropertyType(Base::XMLReader &reader, const char *TypeName, App::Property *prop)
+{
+    // Migrate Material from App::PropertyMap to App::PropertyLink
+    if (!strcmp(TypeName, "App::PropertyMap")) {
+        App::PropertyMap v;
+        v.Restore(reader);
+        if (v.getSize()) {
+            auto msg = "'%s': Migrating 'Material' property to 'App::PropertyLink'\n";
```

Comment:
Why do you output a warning? No other handleChangedPropertyType I find in Part or PartDesign does this and I also don't see the need since it only rasises unnecessary questions for the user.

Reply:
Outputting a warning for such a migration is a recurring pattern in other parts of FreeCAD. See here for instance:
https://github.com/FreeCAD/FreeCAD/blob/master/src/Mod/Draft/draftobjects/pointarray.py#L160
https://github.com/FreeCAD/FreeCAD/blob/master/src/Mod/Arch/ArchIFC.py#L437
https://github.com/FreeCAD/FreeCAD/blob/master/src/Mod/Arch/ArchReference.py#L104
It is also shown as an example in the documentation:
https://wiki.freecadweb.org/Scripted_objects_migration  (see Method 3 example, for instance).

Anyway, I'm totally agnostic about that, so no problem to remove the warning.